In [16]:
import sys, runpy

argv_backup = sys.argv
sys.argv = [
    "train.py",
    "--data-file", "processed_data/training_data_lincs2020_final.pkl",
    "--output-dir", "models/quick_test",
    "--fold", "0",
    "--epochs", "10",
    "--batch-size", "128",
]
runpy.run_path("drugreflector_train/train.py", run_name="__main__")
sys.argv = argv_backup  # 还原

# python train.py \
#     --data-file processed_data/training_data_lincs2020_final.pkl \
#     --output-dir models/quick_test \
#     --fold 0 \
#     --epochs 10 \
#     --batch-size 128


🧬 DRUGREFLECTOR TRAINING
  Based on Science 2025 paper
  Training on LINCS 2020 dataset

📋 Training Configuration:
  Mode: Single fold (0)
  Data file: processed_data/training_data_lincs2020_final.pkl
  Output dir: models/quick_test
  Epochs: 10
  Batch size: 128
  Initial LR: 0.0139
  Focal γ: 2.0
  Device: auto
  Workers: 4

📂 Loading Training Data
  Loading from: processed_data\training_data_lincs2020_final.pkl
  ✓ Data loaded successfully
  Keys: ['X', 'y', 'folds', 'sample_meta', 'metadata', 'gene_names', 'compound_names', 'pert_to_idx']

📊 Data Summary:
  Samples: 258,295
  Features: 978
  Compounds: 6,675
  Unique folds: [0 1 2]

🚀 DrugReflector Trainer Initialized
  Device: cuda
  Initial LR: 0.0139
  Min LR: 1e-05
  Weight Decay: 1e-05
  Focal γ: 2.0
  Batch size: 128
  Epochs: 10
  Workers: 4

📊 Training Fold 0
  Total samples: 258,295
  Total compounds: 6,675
  Gene features: 978

🔧 Preprocessing signatures...

📊 Clipping and normalizing signatures...
   Clip range: (-2, 2)

KeyboardInterrupt: 

In [12]:
import sys, runpy
from pathlib import Path

argv_backup = sys.argv
sys.argv = [
    "train.py",
    "--data-file", "processed_data/training_data_lincs2020_final.pkl",
    "--compound-info", "datasets/LINCS2020/compoundinfo.txt",
    "--fold", "0",
]

# 自动加入脚本所在目录到 sys.path
script_path = Path("drugreflector_twotower/train.py").resolve()
sys.path.append(str(script_path.parent))

runpy.run_path(script_path, run_name="__main__")

sys.argv = argv_backup


Unexpected exception formatting exception. Falling back to standard exception


Traceback (most recent call last):
  File "d:\ProgramData\anaconda3\envs\chemprop\Lib\site-packages\IPython\core\interactiveshell.py", line 3577, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "C:\Users\11234\AppData\Local\Temp\ipykernel_26428\3445058657.py", line 16, in <module>
    runpy.run_path(script_path, run_name="__main__")
  File "<frozen runpy>", line 291, in run_path
  File "<frozen runpy>", line 98, in _run_module_code
  File "<frozen runpy>", line 88, in _run_code
  File "E:\科研\Models\drugreflector\drugreflector_twotower\train.py", line 16, in <module>
    from trainer import TwoTowerTrainer
  File "E:\科研\Models\drugreflector\drugreflector_twotower\trainer.py", line 18, in <module>
    from models import TowTowerModel
  File "E:\科研\Models\drugreflector\drugreflector_twotower\models.py", line 13, in <module>
    from chemprop import nn as cnn
  File "d:\ProgramData\anaconda3\envs\chemprop\Lib\site-packages\chemprop\__init__.py", line 1, in <module>

In [3]:
$python train.py \
    --data-file processed_data/training_data_lincs2020_final.pkl \
    --compound-info datasets/LINCS2020/compoundinfo.txt \
    --fold 0

SyntaxError: invalid syntax (3033893754.py, line 1)

In [ ]:
from chemprop import nn as cnn


AttributeError: partially initialized module 'torchvision' has no attribute 'extension' (most likely due to a circular import)

In [4]:
import torch

torch.cuda.is_available()

True

#!/bin/bash
#SBATCH -J twotower-train
#SBATCH -p gpu_4l
#SBATCH -N 1 
#SBATCH -o ../err_out/twotower_%j.out
#SBATCH -e ../err_out/twotower_%j.err
#SBATCH --no-requeue
#SBATCH -A tangc_g1
#SBATCH --qos=tangcg4c
#SBATCH --gres=gpu:1
#SBATCH --overcommit
#SBATCH --mincpus=30

source /appsnew/source/Tensorflow2.11_Keras2.11.0_keras-nightly2.13.0_torch1.13.1_py3.9.4_cuda11.4.4.sh
unset LD_LIBRARY_PATH

python -u ../drugreflector_twotower/train.py \
    --data-file ../datasets/training_data_lincs2020_final.pkl \
    --compound-info ../datasets/LINCS2020/compoundinfo_beta.txt \
    --output-dir ../models/twotower_ensemble \
    --all-folds \
    --chem-hidden-dim 512 \
    --fusion-method concat \
    --mpnn-depth 3 \
    --epochs 50 \
    --batch-size 256 \
    --learning-rate 0.0139 \
    --use-3d